[CX-568] Add docs for balance, contracts and credit-transactions public api (#291)
cobalthq/cobalt-api-docs: versions/v2/content/credit-transactions.md

```sh
curl -X GET "https://api.us.cobalt.io/credit-transactions" \
  -H "Accept: application/vnd.cobalt.v2+json" \
  -H "Authorization: Bearer YOUR-PERSONAL-API-TOKEN" \
  -H "X-Org-Token: YOUR-V2-ORGANIZATION-TOKEN"
```

> The above command returns JSON structured like this:

```json
{
  "pagination": {
    "next_page": "/resource?cursor=123asdzxc",
    "prev_page": "/resource?cursor=123asdzxc"
  },
  "data": [
    {
      "resource": {
        "recordable_type": "Sales Order",
        "recordable_description": "Adjustment - Timm<>Maria test Call - March 2022 (#PT23657)",
        "amount": -3,
        "transaction_date": "2025-07-11",
        "pending": false
      }
    }
  ],
  "meta": {
    "current_balance": 73921
  }
}
```

This endpoint retrieves a list of credit transactions that belong to the organization specified in the `X-Org-Token` header.

### HTTP Request

`GET https://api.us.cobalt.io/credit-transactions`

### URL Parameters

| Parameter          | Default | Description  |
|--------------------|---------|-----------------|
| `cursor`      | N/A     | Used for [pagination](./#pagination). Example: `https://api.us.cobalt.io/credit-transactions?cursor=a1b2c3d4` |

> | `limit`       | `10`    | If specified, returns only a specified amount of targets. Example: `https://api.us.cobalt.io/credit-transactions?limit=5`|

Reviewer: ```suggestion
| `limit`       | `10`    | If specified, returns only the specified number of transactions. Example: `https://api.us.cobalt.io/credit-transactions?limit=5`|
```
